plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -426,5 +442,37 @@ def temp_release_docs
         end
       end
     end
+
+    def finish_blue_green_deployment(release, release_docs)
+      switch_blue_green_service(release_docs)
+      previous = release.previous_successful_release
+      if previous && previous.blue_green_color != release.blue_green_color
+        previous.release_docs.each { |d| delete_blue_green_resources(d) }
+      end
+    end
+
+    def switch_blue_green_service(release_docs)
+      release_docs.each do |release_doc|
```

Comment:
We should select only `release_docs` having `kubernetes_role.blue_green` to perform this switch. All non-blue/green services were already created in [release_doc.deploy](https://github.com/zendesk/samson/pull/2473/files#diff-026ab5a569538b3fed9f467d2b8acbf3R372)